app/code/Magento/Catalog/Block/Adminhtml/Product/Edit/Button/Back.php

```diff
@@ -17,9 +17,21 @@ public function getButtonData()
     {
         return [
             'label' => __('Back'),
-            'on_click' => sprintf("location.href = '%s';", $this->getUrl('*/*/')),
+            'on_click' => sprintf("location.href = '%s';", $this->getBackUrl()),
             'class' => 'back',
             'sort_order' => 10
         ];
     }
+    /**
+     * Get URL for back
+     *
+     * @return string
+     */
+    public function getBackUrl()
```

Comment:
Hi, @ravi-chandra3197, due to [Magento backward-compatible guide](http://devdocs.magento.com/guides/v2.2/contributor-guide/backward-compatible-development/) we can't create new one public methods. Please make this method private.